app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TweetController < ApplicationController
+
+    get '/tweets' do
+        if !!session[:id]
+            @user = User.find(session[:id])
+            @tweets = Tweet.all
+            erb :'/tweets/index'
+        else
+            redirect '/login'
+        end
+    end
+
+    get '/tweets/new' do
+        if !!session[:id]
+            @user = User.find(session[:id])
+            erb :'/tweets/new'
+        else
+            redirect '/login'
+        end
+    end
+
+    post '/tweets' do
+        @user = User.find(params[:user_id])
+        @tweet = Tweet.new(params)
+        if !@tweet.save
+            redirect '/tweets/new'
+        else
+            @tweet.save
+            redirect '/tweets'
+        end
+    end
+
+    get '/tweets/:id' do
+        if !!session[:id]
+            @user = User.find(session[:id])
+            @tweet = Tweet.find(params[:id])
+            erb :'/tweets/show'
+        else
+            redirect '/login'
+        end
+    end
+
+    get '/tweets/:id/edit' do
+        if !!session[:id]
+            @user = User.find(session[:id])
+            @tweet = Tweet.find(params[:id])
+            erb :'/tweets/edit'
+        else
+            redirect '/login'
+        end
+    end
+
+    post '/tweets/:id' do
```

Comment:
this should be a put/patch statement. You will need to do a hidden tag in the form that handles a `put or patch` method.